virttest/qemu_vm.py

```diff
@@ -1293,6 +1293,32 @@ def _get_pci_bus(devices, params, dtype=None, pcie=False):
                     pass
             return {'aobject': params.get('pci_bus', 'pci.0')}
 
+        def add_pci_controllers(devices, params):
+            """
+            Insert pci controllers into qcontainer by order.
+
+            :param devices: qcontainer object
+            :param params: test params
+            """
+            def sort_key(dev):
+                """
+                Function used to sort pcic list
+                """
+                order_pcics = ['ioh3420', 'x3130-upstream',
+                               'x3130', 'i82801b11-bridge',
+                               'pci-bridge']
+                try:
+                    return order_pcics.index(dev.get_param('driver'))
+                except ValueError:
+                    return -1
+
+            pcics = []
+            for pcic in params.objects("pci_controllers"):
+                dev = devices.pcic_by_params(pcic, params.object_params(pcic))
+                slot = len([_ for _ in pcics if _.get_param('driver') == 'ioh3420'])
+                dev.set_param('slot', hex(slot))
```

Comment:
If driver is not ioh3420, slot will not be assigned value, will this line hit error therefore?